Core/GDCore/Project/Variable.cpp

```diff
@@ -66,99 +105,106 @@ const Variable& Variable::GetChild(const gd::String& name) const {
   auto it = children.find(name);
   if (it != children.end()) return *it->second;
 
-  isStructure = true;
+  type = TYPES::STRUCTURE;
   children[name] = std::make_shared<gd::Variable>();
   return *children[name];
 }
 
 void Variable::RemoveChild(const gd::String& name) {
-  if (!isStructure) return;
+  if (type == TYPES::STRUCTURE) return;
   children.erase(name);
-  isStructure = !children.empty();
+
+  // If the structure is empty, make it a default empty variable again.
+  if (children.empty()) {
+    type = TYPES::NUMBER;
+    value = 0;
+  }
 }
 
 bool Variable::RenameChild(const gd::String& oldName,
                            const gd::String& newName) {
-  if (!isStructure || !HasChild(oldName) || HasChild(newName)) return false;
+  if (type != TYPES::STRUCTURE || !HasChild(oldName) || HasChild(newName))
+    return false;
 
   children[newName] = children[oldName];
   children.erase(oldName);
 
   return true;
 }
 
-void Variable::ClearChildren() {
-  if (!isStructure) return;
-  children.clear();
-}
+bool Variable::HasIndex(const size_t index) const {
+  return type == TYPES::ARRAY && childrenList.size() < index;
+};
+
+Variable& Variable::GetIndex(const size_t index) {

```

Comment:
Should this be called `GetAt`? As we're getting something (an "item" or an "element" at some position), but we're not getting an index.

Reply:
GetAt and RemoveAt clash with the variable container macro that converts GetAt to Get and RemoveAt to Remove (Because of overloading)